apps/meteor/client/views/room/lib/getRoomDirectives.ts

```diff
@@ -13,18 +13,18 @@ type getRoomDirectiesType = {
 	roomCanRemove: boolean;
 };
 
-export const getRoomDirectives = (room: IRoom): getRoomDirectiesType => {
+export const getRoomDirectives = (room: IRoom, showingUserId: IUser['_id'], userSubscription?: ISubscription): getRoomDirectiesType => {
```

Comment:
1. I don't like the idea of having functions with 3+ positional parameters instead of using an object for named parameters, it might hurt on refactors and code reviews;
2. If you have to inject so many dependencies in a function, might be better to introduce some companion helpers, like a React hook:

```ts
const useRoomDirectives = () => {
  const room = useRoom(); // consumes `RoomContext`
  const subscription = useRoomSubscription(); // consumes `RoomContext`
  const uid = useUserId(); // consumes `UserContext`
  return useCallback(() => getRoomDirectives({ room, uid, subscription }));
};
```

This example illustrates how we use React contexts as dependency containers.

Reply:
Given the time (release constraints) I preferred to go with option 1, and I'll refactor that. Thanks for the suggestion.